modules/background-downloader/android/src/main/java/expo/modules/backgrounddownloader/BackgroundDownloaderModule.kt

```
@@ -72,32 +87,45 @@ class BackgroundDownloaderModule : Module() {
-    AsyncFunction("startDownload") { urlString: String, destinationPath: String?, promise: Promise ->
+    // `metadata` carries the iOS Live Activity payload; here only `itemId` is used, echoed back in
+    // events so JS can correlate them without a taskId bookkeeping layer. The parameter must exist
+    // on both platforms regardless — JS always passes three arguments.
+    AsyncFunction("startDownload") { urlString: String, destinationPath: String?, metadata: Map<String, Any?>?, promise: Promise ->
       try {
-        val taskId = startDownloadInternal(urlString, destinationPath)
+        val taskId = synchronized(stateLock) {
+          startDownloadLocked(urlString, destinationPath, metadata?.get("itemId") as? String)
+        }
         promise.resolve(taskId)
       } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)